cb-check/main.ml

```diff
@@ -0,0 +1,52 @@
+let read_all ic =
+  let b = Buffer.create 80 in
+  let rec loop () =
+    let line = input_line ic in
+    Buffer.add_string b (line ^ "\n");
+    loop ()
+  in
+  try loop ()
+  with End_of_file ->
+    close_in ic;
+    Buffer.contents b
+
+let validate l =
+  let aux (str, _) =
+    match Yojson.Safe.from_string str with
+    | j -> (
+        match Cb_schema.S.of_json j with
+        | s -> Some s
+        | exception Invalid_argument s ->
+            Format.eprintf
+              "Some valid json didn't conform to the schema with error: %S\n\
+               The json: %a\n"
+              s
+              (Yojson.Safe.pretty_print ~std:false)
+              j;
+            None)
+    | exception Yojson.Json_error s ->
+        Format.eprintf "\x1b[91mJson parsing failure, please report: \x1b[0m%s"
+          s;
+        exit 1
+  in
+  match List.filter_map aux l with
+  | [] ->
+      Format.printf "No schema-valid results were parsed.\n";
+      exit 1
+  | validated ->
+      print_endline "Correctly parsed some benchmarks:";
+      Cb_schema.S.merge [] validated
+      |> List.map Cb_schema.S.to_json
+      |> List.iter (fun j ->
+             Format.printf "%s@;" (Cb_schema.Json.pp_to_string () j))
```

Comment:
Nitpick: using `@.` for line breaks might work better with `Format` since it also forces a flush (while `@;` may choose not to break but [output spaces](https://v2.ocaml.org/api/Format.html#VALprint_break), and in above lines you use `\n` which doesn't cause Format to flush its internal buffer to sdout). It would also be nice to only use Format rather than mix it with `print_endline` (again, for buffering reasons)

Reply:
According to https://v2.ocaml.org/releases/5.0/api/Format.html#VALprint_newline, `@.` is equivalent to `print_newline`, which isn't recommended. Do you think we should break from their recommendations?
I agree that the `print_endline` is misplaced though

Reply:
Yes I think so! The Format documentation was written with the intent that one would use Format to pretty-print large ASTs, and in that context, using `@.` is wrong since it signals the end of the document (to be flushed on stdout) rather than a recommended line-break in the syntax of the AST. But when using [Format as a better Printf](https://sherlocode.com/?q=%40.%22) it makes sense to output small lines where `@.` is perfectly appropriate to end the print :) (we want Format to output immediately, and the following lines should not be influenced by an earlier formatting bug, for example a non-closed `@[` leading to absurd indentation)